addons/payment_custom/models/payment_provider.py

```diff
@@ -38,6 +38,14 @@ def _compute_view_configuration_fields(self):
             'show_cancel_msg': False,
         })
 
+    def _get_provider_removal_values(self):
+        res = super()._get_provider_removal_values()
+
+        if self.code == 'custom':
+            res['custom_mode'] = False
```

Comment:
```suggestion
        res['custom_mode'] = False
```

`self` is empty here.  IMHO, you can always set custom_mode to False, it won't be a problem.